internal/app/components/timelines/timelines.go

```diff
@@ -22,12 +22,11 @@ func newTimelineId() TimelineId {
 	return TimelineId(xid.New().String())
 }
 
-// TODO: use cr parameter when creating Timeline
-func NewTimeline(name TimelineName, cr *courses.CourseRecords) *Timeline {
+func NewTimeline(name TimelineName, tr []*terms.TermRecord) *Timeline {
```

Comment:
The new timeline will still have a slice of references to the course history's term records. In order to create a new timeline, you need to perform deep copy instead of shallow copy. Also, could you name `tr` to something like `courseHistory` or `history` so it's easier for the caller to understand what it's supposed to take